libcudacxx/include/cuda/__driver/driver_api.h

```diff
@@ -324,6 +325,20 @@ _CCCL_HOST_API void __memsetAsync(void* __dst, _Tp __value, size_t __count, ::CU
   }
 }
 
+_CCCL_HOST_API inline ::cudaError_t
+__memRetainAllocationHandleNoThrow(::CUmemGenericAllocationHandle* __handle, void* __addr)
+{
+  static auto __driver_fn = _CCCLRT_GET_DRIVER_FUNCTION(cuMemRetainAllocationHandle);
+  return static_cast<::cudaError_t>(__driver_fn(__handle, __addr));
+}
+
+_CCCL_HOST_API inline ::cudaError_t
+__memGetAllocationPropertiesFromHandleNoThrow(::CUmemAllocationProp* __prop, ::CUmemGenericAllocationHandle __handle)
+{
+  static auto __driver_fn = _CCCLRT_GET_DRIVER_FUNCTION(cuMemGetAllocationPropertiesFromHandle);
+  return static_cast<::cudaError_t>(__driver_fn(__prop, __handle));
+}
+
 _CCCL_HOST_API inline ::cudaError_t __mempoolCreateNoThrow(::CUmemoryPool* __pool, ::CUmemPoolProps* __props)
```

Comment:
Here too while we are at it
```suggestion
[[nodiscard]] _CCCL_HOST_API inline ::cudaError_t __mempoolCreateNoThrow(::CUmemoryPool* __pool, ::CUmemPoolProps* __props)
```